### ISSUE TITLE
Real Name

### ISSUE THREAD

**Issue by tschet** [EPSCoR/ERCore#23]
Within the ERCore module, there is some code relating to displaying users real name rather than their site username (function get_real_name).
I would like to see us switch to using the https://www.drupal.org/project/realname instead of a custom function. The realname module changes the site-wide display of usernames to their "real name" as defined within the module settings.  It integrates nicely with LDAP and other authentication modules, as well as Views, Features, etc. I think this is a simpler solution than maintaining a custom function and gives more functionality. 
Just a thought.

**Comment by iserna** [comment]
Good thought @tschet.  Have you looked into editing your local site to include or update the module to use the Real Name?  If you do, can you create a branch, that includes the changes, and share it with everyone?  We can then download it and test it out on our local systems and give feedback or add/comment on it?

Chris and I briefly talked about this and he does not recall this module being in existance or completed back when the ER Core module was initially be planned and developed.  Though, he too thought it would be more efficient and useful than the custom function that's currently in place.  

Sorry for the delay in responding to this:)

**Comment by tschet** [comment]
I pushed a "realname" branch to https://github.com/tschet/ERCore-3.1. I don't have permission to push anything here. The realname branch isn't a final version, but when installing from scratch it should work. There is an hook update which worked when I updated an existing site from the default version to "realname". I'm not confident that will work cleanly on sites with any modifications however. It seems close enough to share and discuss.

**Comment by cjallen-epscor** [comment]
@tschet - strange. I didn't know that you weren't a member of the ERCore repo. I just added you to this repo as a member.

**Comment by tschet** [comment]
Thank you.

Douglas Tschetter
701.202.6191

**Comment by cjallen-epscor** [comment]
@tschet - i attempted to load this Realname version (https://github.com/tschet/ERCore-3.1/tree/realname) into a new sandbox server locally and I get the following error:

```
Fatal error: require_once(): Failed opening required
'../sites/all/modules/er/plugins/selection/realname.inc' 

(include_path='.:/Applications/MAMP/bin/php/php5.6.2/lib/php') in 

/../sites/all/modules/ctools/includes/plugins.inc on line 477
```

**Comment by cjallen-epscor** [comment]
@tschet - Disregard that last post. I realized I had an old branch.

**Comment by cjallen-epscor** [comment]
@tschet - i did find one error - 
After loading the real name branch I attempted to add attendee data on an EE.
The Calendar Event Title does not load correctly in the EE.  
![screen shot 2016-04-27 at 12 26 40 pm](https://cloud.githubusercontent.com/assets/8905892/14863571/7713f914-0c73-11e6-8fb8-a8096e2022a2.png)

I am running entity_reference_prepulate 1.4 on the install with real name (left page in image) - which may be due to this https://github.com/EPSCoR/ERCore-3.1/issues/29 ?

**Comment by tschet** [comment]
I'm not sure if #29 would apply, but it could be. It seems to me that I had to modify something related to autocomplete related to users. It could be that I was to general in my code and my fix for users is interfering with node autocomplete.

**Comment by cjallen-epscor** [comment]
Yeah - I tried adding an external collaborator to a collaboration and 
1) the user lookup field works correctly 
2) however, clicking on 'auto-fill fields below' does nothing

**Comment by tschet** [comment]
That sounds right. I have had that working intermittently.  When I have time I'll look at that again.

Douglas T

**Comment by khuffman** [comment]
Adding this comment to refer that this issue is related to https://github.com/EPSCoR/ERCore-3.1/issues/46
